vnext/JSI/Shared/ChakraCoreRuntime.h

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#ifdef CHAKRACORE
+
+#include "ChakraCoreRuntimeArgs.h"
+#include "ChakraRuntime.h"
+
+#include "ChakraCoreDebugger.h"
+
+#include <memory>
+#include <string>
+
+namespace Microsoft::JSI {
+
+class ChakraCoreRuntime : public ChakraRuntime {
+ public:
+  ChakraCoreRuntime(const std::shared_ptr<ChakraCoreRuntimeArgs> &args) noexcept;
+  ~ChakraCoreRuntime() noexcept;
+
+#pragma region Functions_inherited_from_Runtime
+
+  facebook::jsi::Value evaluatePreparedJavaScript(
+      const std::shared_ptr<const facebook::jsi::PreparedJavaScript> &preparedJs) override;
+
+  std::string description() override;
+
+ private:
+  // We override the following because ChakraCore (unlike Chakra) offers
+  // weak reference semantics.
+  facebook::jsi::WeakObject createWeakObject(const facebook::jsi::Object &obj) override;
```

Comment:
>override [](start = 79, length = 8)

Should these be protected instead of private?

Reply:
Nothing is inheriting from ChakraCoreRuntime right now so I don't think there is any reason for making members protected. If we need change this to protected in the future, we'll do it then.

---
In reply to: [345443262](https://github.com/microsoft/react-native-windows/pull/3650#discussion_r345443262) [](ancestors = 345443262)